### PR TITLE
[Logs] Clean core worker logs 

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Clean up corer worker logs

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/transport/direct_actor_transport.cc [hunk 1]
```diff
@@ -138,7 +138,8 @@ void CoreWorkerDirectActorTaskSubmitter::ConnectActor(const ActorID &actor_id,
   if (num_restarts < queue->second.num_restarts) {
     // This message is about an old version of the actor and the actor has
     // already restarted since then. Skip the connection.
-    RAY_LOG(INFO) << "Skip actor that has already been restarted, actor_id=" << actor_id;
+    RAY_LOG(INFO) << "Skip actor creation that has already been restarted, actor_id="
```
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_actor_transport.cc:141]: should be `Skip actor connection`?

**Review by ericl — APPROVED** [review]

File: src/ray/core_worker/transport/direct_actor_transport.cc [hunk 2]
```diff
@@ -188,7 +191,8 @@ void CoreWorkerDirectActorTaskSubmitter::DisconnectActor(
   if (num_restarts <= queue->second.num_restarts && !dead) {
     // This message is about an old version of the actor that has already been
     // restarted successfully. Skip the message handling.
-    RAY_LOG(INFO) << "Skip actor that has already been restarted, actor_id=" << actor_id;
+    RAY_LOG(INFO) << "Skip actor creation that has already been restarted, actor_id="
```
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_actor_transport.cc:194]: disconnection

File: src/ray/core_worker/transport/direct_actor_transport.h [hunk 3]
```diff
@@ -416,8 +416,8 @@ class ActorSchedulingQueue : public SchedulingQueue {
         pool_(pool),
         is_asyncio_(is_asyncio) {
     if (is_asyncio_) {
-      RAY_LOG(INFO) << "Setting actor as async with max_concurrency=" << max_concurrency
-                    << ", creating new fiber thread.";
+      RAY_LOG(DEBUG) << "Setting actor as async with max_concurrency=" << max_concurrency
```
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_actor_transport.h:419]: This log only prints once when initialized and max_concurrency is helpful(i think), maybe INFO is ok?

File: src/ray/core_worker/transport/direct_task_transport.cc [hunk 4]
```diff
@@ -44,7 +44,7 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
       // https://docs.google.com/document/d/1EAWide-jy05akJp6OMtDn58XOK7bUyruWMia4E-fV28/edit?usp=sharing
       auto actor_id = task_spec.ActorCreationId();
       auto task_id = task_spec.TaskId();
-      RAY_LOG(INFO) << "Creating actor via GCS actor id = : " << actor_id;
+      RAY_LOG(DEBUG) << "Creating actor via GCS actor id = : " << actor_id;
```
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_task_transport.cc:47]: I'm not sure about this. The actor/worker creation is not of high frequency and sometimes it helps us to see what happened when the actor didn't come up successfully. Maybe we keep its level unchanged?

Comment (rkooo567) [reply src/ray/core_worker/transport/direct_task_transport.cc:47]: Hmm yeah I thought there should be the same duplicated logs in the upper layer (like core_worker.cc), but it seems like it doesn't. Let me revert it.

File: src/ray/core_worker/transport/direct_actor_transport.h (hunk 3 above)
Comment (rkooo567) [reply src/ray/core_worker/transport/direct_actor_transport.h:419]: sgtm.

**Review by WangTaoTheTonic — APPROVED** [review]
LGTM!

**Comment by rkooo567** [comment]
The failing test is flaky in the master

**Pull request merged**